File: src/code/cpp/sorting_algorithms/selection_sort.cpp
#include <vector>

using namespace std;


void SelectionSort(vector<int>& arr) {
    int n = arr.size();

    for (int i = 0; i < n; ++i) {
        int min_i = i;

        for (int j = i + 1; j < n; ++j) {
            if (arr[j] < arr[min_i]) {
                min_i = j;
            }
        }

        if (min_i != i) {
            swap(arr[i], arr[min_i]);
        }
    }
}

File: src/code/ruby/sorting_algorithms/selection_sort.rb
def selection_sort(arr)
  n = arr.size

  n.times do |i|
    min_i = i

    ((i + 1)...n).each do |j|
      min_i = j if arr[j] < arr[min_i]
    end

    arr[i], arr[min_i] = arr[min_i], arr[i] if min_i != i
  end
end
